content/docs/user-guide/checkpoints.md

```diff
@@ -0,0 +1,195 @@
+# Checkpoints
+
+ML checkpoints are an important part of deep learning because ML engineers like
+to save the model files at checkpoints during a training process and return back
+when metrics start diverging.
+
+_Note: This is not a continuation of the previous Getting Started sections._
+
+With checkpoints in DVC, you can track every training epoch, return to any
+previous checkpoint, see when metrics converge or when the learning rate needs
+to be adjusted, and even recover model weights, parameters, and code. You also
+have the ability to resume training from previous checkpoints. When you adjust
+parameters and code, DVC tracks those changes for you.
+
+## Adding checkpoints
+
+When you want to add a checkpoint to your training process, there are only a
+couple of things you need to setup. You can find the code we use here in
+[this GitHub repo](https://github.com/iterative/dvc-checkpoints-mnist).
+
+In your `dvc.yaml` file, add `checkpoint: true` to your `outs` under the model
+file. This adds checkpoints at the pipeline level. It should look something like
+this.
+
+```yaml
+stages:
+  train:
+    cmd: python train.py
+    deps:
+      - train.py
+    outs:
+      - model.pt:
+          checkpoint: true
+    live:
+      logs:
+        summary: true
+        html: true
+```
+
+You can also add checkpoints by running the command
+`dvc stage add -n name_of_stage --checkpoints model.pt`. This will create a new
+stage with checkpoints enabled.
```

Comment:
So if they have an existing stage, the only way to add checkpoints is to manually update the `dvc.yaml` with `checkpoint: true`?

Reply:
You can rewrite stages with `stage add -f`. But anyway, what about making this a very shot note about `stage add` since I think we want to encourage users to edit dvc.yaml manually? Something like

"See also `dvc stage add --checkpoint` to generate the stage from CLI."